aiida_nanotech_empa/utils/gaussian_spin_wc_postprocess.py

```diff
@@ -0,0 +1,190 @@
+
+        print("{:>10} {:>16} {:>6} {:>16} {:>6}".format(
+            "i_orb", "E(up) (eV)", "occ(up)", "E(down) (eV)", "occ(down)"))
+
+        max_homo = np.max([mo_e[0, homo_s0], mo_e[1, homo_s1]])
+        min_lumo = np.min([mo_e[0, homo_s0 + 1], mo_e[1, homo_s1 + 1]])
+
+        for i_orb in range(i_end, i_start, -1):
+            occ0 = 1 if i_orb <= homo_s0 else 0
+            occ1 = 1 if i_orb <= homo_s1 else 0
+            print("{:10d} {:16.4f} {:6d} {:16.4f} {:6d}".format(
+                i_orb, mo_e[0, i_orb], occ0, mo_e[1, i_orb], occ1))
+
+        print("GAP alpha: {:.4f} eV".format(mo_e[0, homo_s0 + 1] -
+                                            mo_e[0, homo_s0]))
+        print("GAP beta:  {:.4f} eV".format(mo_e[1, homo_s1 + 1] -
+                                            mo_e[1, homo_s1]))
+        print("GAP eff:   {:.4f} eV".format(min_lumo - max_homo))
+
+
+def print_out_params(out_params):
+
+    if out_params["spin_expectation_values"]:
+        spin_dict = out_params["spin_expectation_values"][-1]
+        s_ideal = (out_params['mult'] - 1) / 2
+        print("S**2: {0:.3f}, ideal: {1:.1f}".format(spin_dict['S**2'],
+                                                     s_ideal * (s_ideal + 1)))
+        print("S:    {0:.3f}, ideal: {1:.1f}".format(spin_dict['S'], s_ideal))
+    print()
+
+    if "moenergies" in out_params:
+        print("MO energies:")
+        print_orb_energies(out_params)
+
+
+def print_no_analysis(natorb_dict, prop_dict, n_orb=4):
+
+    no_occs = natorb_dict['no_occs']
+    no_occs_sp = natorb_dict['no_occs_sp']
+
+    i_start = prop_dict['homos'][0] - n_orb
+    i_end = prop_dict['homos'][0] + n_orb
+
+    print("{:>20} {:>16} {:>16}".format("i_no", "occ", "sp. proj. occ"))
+
+    for i_no in range(i_end, i_start, -1):
+        print("{:>20} {:16.4f} {:16.4f}".format(i_no, no_occs[i_no],
+                                                no_occs_sp[i_no]))
+    print("---------------------------------------------------------")
+    print("{:>20} {:16.4f} {:16.4f}".format("Standard num odd:",
+                                            natorb_dict['std_num_odd'],
+                                            natorb_dict['std_num_odd_sp']))
+    print("{:>20} {:16.4f} {:16.4f}".format("HG num odd:",
+                                            natorb_dict['hg_num_odd'],
+                                            natorb_dict['hg_num_odd_sp']))
+
+
+def plot_spin_wc(cube_image_folder):
+    #pylint: disable=protected-access
+    folder_path = cube_image_folder._repository._get_base_folder().abspath
+    image_names = cube_image_folder.list_object_names()
+    for imag_name in image_names:
+        if "spin" in imag_name and "z+" in imag_name and "iv0.010" in imag_name:
+            pil_image = Image.open(folder_path + "/" + imag_name)
+            plt.imshow(pil_image)
+            plt.title(imag_name)
+            plt.axis('off')
+            plt.show()
+
+
+def make_report(wc_node):
```

Comment:
Here I would rather generate a string and return it, printing seems to be not very useful.

Reply:
The report contains pictures as well, so at the moment I would keep it like this. In the future we can adapt for different use-cases (e.g. if we want to make an aiidalab app).